src/client/matrixMapper.ts

```diff
@@ -0,0 +1,187 @@
+import {
+  BatchErrorResponse,
+  BatchResponse,
+  Coords,
+  LocationRequest,
+  TimeFilterFastRequest,
+  TimeFilterFastRequestProperty,
+  TimeFilterFastResponse,
+  TimeFilterRequest,
+  TimeFilterRequestProperty,
+  TimeFilterResponse,
+
+} from '../types';
+import {
+  TimeFilterFastManyToManyMatrixRequest, TimeFilterFastManyToManyMatrixResponse, TimeFilterManyToManyMatrixRequest, TimeFilterManyToManyMatrixResponse,
+} from '../types/timeFilterMatrix';
+
+function validateMaxSearchLimit(MAX_SEARCHES_LIMIT: number, maxSearchesPerRequest: number | undefined) {
+  const maxSearches = maxSearchesPerRequest || MAX_SEARCHES_LIMIT;
+  if (maxSearches > MAX_SEARCHES_LIMIT) {
+    throw new Error(`Max number of searches is ${MAX_SEARCHES_LIMIT}`);
+  }
+  return maxSearches;
+}
+
+function generateLocationsForChunk(indexFrom: number, coordsFrom: Coords, chunk: Coords[], chunkOffset: number): LocationRequest[] {
+  return [
+    { id: `from-${indexFrom}`, coords: coordsFrom },
+    ...chunk.map((coordsTo, iTo) => ({ id: `to-${iTo + chunkOffset}`, coords: coordsTo })),
+  ];
+}
+
+function generateRequestsFromChunks<R>(body: {
+  maxSearchesPerRequest: number | undefined
+  coordsFrom: Coords[]
+  coordsTo: Coords[]
+}, makeRequest: (params: {
+  iFrom: number,
+  coordsFrom: Coords,
+  chunk: Coords[],
+  chunkOffset: number,
+}) => R): R[] {
+  const maxSearchesPerRequest = validateMaxSearchLimit(100_000, body.maxSearchesPerRequest);
+  const requests: R[] = [];
+  body.coordsFrom.forEach((coordsFrom, iFrom) => {
+    for (let index = 0; index < body.coordsTo.length; index += maxSearchesPerRequest) {
+      const chunk = body.coordsTo.slice(index, index + maxSearchesPerRequest);
+      requests.push(makeRequest({
+        iFrom, coordsFrom, chunk, chunkOffset: index,
+      }));
+    }
+  });
+  return requests;
+}
+
+const parseSingleIdToNumber = (locationId: string) => Number.parseInt(locationId.substring(3), 10);
+
+function getPrimaryIndex(searchId: string) {
+  const rgx = /.*-(\d+)-.*(\d+)/;
+  const rgxMatch = rgx.exec(searchId);
+  if (rgxMatch) {
+    const primaryIndex = Number.parseInt(rgxMatch[1], 10);
+    if (!Number.isNaN(primaryIndex)) return primaryIndex;
+  }
+
+  return false;
```

Comment:
I would change the return type to just number here and this return value to -1. This way its more clear behaviour for anyone not using types.